tool/templates/c/enum.h.jinja

```diff
@@ -5,4 +5,4 @@ enum {{ ty_name }} {
   {%- endfor %}
 } {%- if !is_for_cpp %} {{ ty_name }} {%- endif %};
 
-
+typedef struct {{ ty_name }}_option {union { {{ty_name}} ok; }; bool is_ok; } {{ ty_name }}_option;
```

Comment:
Question: why the `union`? To avoid calling the default constructor?

Reply:
There are potential safety issues around strict aliasing etc if it is not marked as a union. The field is allowed to be uninit, we mark it as such with a union.